src/Features/CSharp/Portable/EditAndContinue/StatementSyntaxComparer.cs

```diff
@@ -192,6 +191,8 @@ internal enum Label
             LocalDeclarationStatement,         // tied to parent
             LocalVariableDeclaration,          // tied to parent
             LocalVariableDeclarator,           // tied to parent
+            DeconstructionDeclarationStatement, // tied to parent
+            VariableComponentAssignment, // tied to parent
```

Comment:
For full EnC support we'll need to label each ```SingleVariableDesignation``` node (which would be considered the *declarator* node of the variable) and all it's parent nodes. Similarly to ```LocalVariableDeclarator```.